examples/chef/common/chef-channel-manager.cpp

```diff
@@ -0,0 +1,210 @@
+    lineup.lineupInfoType = chip::app::Clusters::Channel::LineupInfoTypeEnum::kMso;
+
+    return aEncoder.Encode(lineup);
+}
+
+CHIP_ERROR ChefChannelManager::HandleGetCurrentChannel(app::AttributeValueEncoder & aEncoder)
+{
+    return aEncoder.Encode(mCurrentChannel);
+}
+
+void ChefChannelManager::HandleChangeChannel(CommandResponseHelper<ChangeChannelResponseType> & helper,
+                                             const chip::CharSpan & match)
+{
+    std::vector<ChannelInfoType> matchedChannels;
+    uint16_t index = 0;
+    for (auto const & channel : mChannels)
+    {
+        // verify if CharSpan matches channel name
+        // or callSign or affiliateCallSign or majorNumber.minorNumber
+        if (isChannelMatched(channel, match))
+        {
+            matchedChannels.push_back(channel);
+        }
+        else if (matchedChannels.size() == 0)
+        {
+            // "index" is only used when we end up with matchedChannels.size() == 1.
+            // In that case, we want it to be the number of non-matching channels we saw before
+            // the matching one.
+            index++;
+        }
+    }
+
+    ChangeChannelResponseType response;
+
+    // Error: Found multiple matches
+    if (matchedChannels.size() > 1)
+    {
+        response.status = chip::app::Clusters::Channel::ChannelStatusEnum::kMultipleMatches;
+        helper.Success(response);
+    }
+    else if (matchedChannels.size() == 0)
+    {
+        // Error: Found no match
+        response.status = chip::app::Clusters::Channel::ChannelStatusEnum::kNoMatches;
+        helper.Success(response);
+    }
+    else
+    {
+        response.status      = chip::app::Clusters::Channel::ChannelStatusEnum::kSuccess;
+        response.data        = chip::MakeOptional(CharSpan::fromCharString("data response"));
+        mCurrentChannel      = matchedChannels[0];
+        mCurrentChannelIndex = index;
+        helper.Success(response);
+    }
+}
+
+bool ChefChannelManager::HandleChangeChannelByNumber(const uint16_t & majorNumber, const uint16_t & minorNumber)
+{
+    // TODO: Insert code here
+    bool channelChanged = false;
+    uint16_t index      = 0;
+    for (auto const & channel : mChannels)
+    {
+        // verify if major & minor matches one of the channel from the list
+        if (channel.minorNumber == minorNumber && channel.majorNumber == majorNumber)
+        {
+            // verify if channel changed by comparing values of current channel with the requested channel
+            if (channel.minorNumber != mCurrentChannel.minorNumber || channel.majorNumber != mCurrentChannel.majorNumber)
+            {
+                channelChanged       = true;
+                mCurrentChannelIndex = index;
+                mCurrentChannel      = channel;
+            }
+        }
+        index++;
+    }
+    return channelChanged;
+}
+
+bool ChefChannelManager::HandleSkipChannel(const int16_t & count)
```

Comment:
replace `const &` for integers with just the integer everywhere. Also probably making count be a int32_t seems to avoid the need for a cast.

Reply:
Actually the original definition is in [channel-delegate.h](https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/channel-server/channel-delegate.h#L44) . Do you suggest to replace them all?